
# Phase I Part 1 — Data Overview
This notebook loads the Airbnb NYC 2019 dataset and reports the structural diagnostics requested for Phase I.


In [1]:

from pathlib import Path
import pandas as pd
from IPython.display import display

DATA_PATH = Path('AB_NYC_2019.csv')
df = pd.read_csv(DATA_PATH)
print(f"Loaded {DATA_PATH.resolve()}")


Loaded /Users/bowen/Desktop/644 teamwork/AB_NYC_2019.csv


In [3]:

print('Shape (rows, columns):', df.shape)
display(df.head(5))


Shape (rows, columns): (48895, 16)


,id,name,host_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365
0,2539,Clean & quiet apt home by the park,2787,John,Brooklyn,Kensington,40.64749,-73.97237,Private room,149,1,9,2018-10-19,0.21,6,365
1,2595,Skylit Midtown Castle,2845,Jennifer,Manhattan,Midtown,40.75362,-73.98377,Entire home/apt,225,1,45,2019-05-21,0.38,2,355
2,3647,THE VILLAGE OF HARLEM....NEW YORK !,4632,Elisabeth,Manhattan,Harlem,40.80902,-73.94190,Private room,150,3,0,NaN,NaN,1,365
3,3831,Cozy Entire Floor of Brownstone,4869,LisaRoxanne,Brooklyn,Clinton Hill,40.68514,-73.95976,Entire home/apt,89,1,270,2019-07-05,4.64,1,194
4,5022,Entire Apt: Spacious Studio/Loft by central park,7192,Laura,Manhattan,East Harlem,40.79851,-73.94399,Entire home/apt,80,10,9,2018-11-19,0.10,1,0


**Note:** `df.shape` confirms the dataset has 48,895 rows and 16 columns; reviewing `df.head()` alongside it helps verify that column meanings and value ranges match the Airbnb NYC 2019 documentation.


In [2]:

df.info()


<class 'pandas.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  str    
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  str    
 4   neighbourhood_group             48895 non-null  str    
 5   neighbourhood                   48895 non-null  str    
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  str    
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     38843 non-n

**Note:** `df.info()` shows most columns are numeric or object types, while fields like `last_review` load as strings and must be converted before any time-based analysis.


In [4]:

missing = df.isna().sum().to_frame(name='missing_count')
missing['missing_pct'] = (missing['missing_count'] / len(df) * 100).round(2)
display(missing)


,missing_count,missing_pct
id,0,0.00
name,16,0.03
host_id,0,0.00
host_name,21,0.04
neighbourhood_group,0,0.00
neighbourhood,0,0.00
latitude,0,0.00
longitude,0,0.00
room_type,0,0.00
price,0,0.00


**Note:** Missing values cluster in `reviews_per_month` and `last_review` (about 10k rows) with minor gaps in `host_name` and `name`, which motivates the minimal filling/dropping strategy in the cleaning notebook.
